### PR TITLE
[Eval] Agent review bench

### PR DESCRIPTION
**Pull request by sarthakaggarwal97**
Branch for CI agent eval. Do not merge.

### CODE REVIEW

**Comment by sarthakaggarwal97** [comment]
## PR Summary

This reworks how `redactClientCommandArgument()` hides sensitive args (AUTH passwords, ACL secrets, CLUSTER SETSLOT tokens, etc.) from the command log and MONITOR output. Instead of eagerly cloning `c

### Walkthrough

This reworks how `redactClientCommandArgument()` hides sensitive args (AUTH passwords, ACL secrets, CLUSTER SETSLOT tokens, etc.) from the command log and MONITOR output. Instead of eagerly cloning `c->argv` into `original_argv` and swapping in `shared.redacted` at write time, the client now carries a `uint32_t redact_arg_bitmap` and consumers (`commandlog.c`, `replicationFeedMonitors`) check `clientCommandArgShouldBeRedacted()` at read time.

The motivation is to skip the argv backup allocation on the hot path for commands that flag args as sensitive — every AUTH/HELLO/CONFIG SET/ACL SETUSER previously forced a full argv copy even when nothing was being rewritten. The bitmap approach is ~free by comparison.

A few things worth flagging. The `>= 32` handling sets bit 0 as a sentinel that means "redact *everything* past 31," so a single call with a high index over-redacts siblings — this shows up for `config.c:837` (`2 + i*2 + 1`), where a CONFIG SET with ~16+ pairs would redact all remaining args, keys included. That's safe-by-default but worth documenting more visibly, since today's code only says "all arguments beyond the bitmap range." Second, the invariant moved from write-time (replacing in `original_argv`) to read-time (every consumer must ask), so any other site that logs or echoes `c->argv` now needs the same check — the two updated in this PR look right, but an audit for future-proofing wouldn't hurt. Finally, there are no new tests covering the bitmap, the `argc < 1` assert, or the `>= 32` sentinel behavior; given the rebase churn (the "extend → revert → redo" sequence on that exact edge case), a TCL test for AUTH/ACL redaction surviving into commandlog output would be reassuring. DCO trailers are present on all commits.

### Maintainer Checklist

- Core review: changed paths suggest `@core-team` review may be needed.

**Review by sarthakaggarwal97 — COMMENTED** [review]
Automated review found 1 issue(s). Highest severity: `medium`.

**Review by sarthakaggarwal97 — COMMENTED** [review]
Automated review found 1 issue(s). Highest severity: `low`.

**Review by sarthakaggarwal97 — COMMENTED** [review]
Automated review found 4 issue(s). Highest severity: `high`.

File: src/networking.c [hunk 1]
```diff
@@ -5854,17 +5856,24 @@ static void backupAndUpdateClientArgv(client *c, int new_argc, robj **new_argv)
...
 void redactClientCommandArgument(client *c, int argc) {
-    backupAndUpdateClientArgv(c, c->argc, NULL);
-    if (c->original_argv[argc] == shared.redacted) {
-        /* This argument has already been redacted */
-        return;
+    serverAssert(argc >= 1);
+    if (argc < 32) {
+        c->redact_arg_bitmap |= (1U << argc);
+    } else {
+        c->redact_arg_bitmap |= 1U;
```
Comment (sarthakaggarwal97) [inline src/networking.c:5875]: Using bit 0 as "redact everything ≥ 32" means that as soon as *any* caller redacts *one* index ≥ 32, `clientCommandArgShouldBeRedacted` returns true for *every* index ≥ 32. That changes behavior for legitimate sparse callers — e.g. `configSetCommand` (src/config.c:837) redacts only the sensitive half of each `k v` pair: `redactClientCommandArgument(c, 2 + i*2 + 1)`. For `CONFIG SET` with ≥16 pairs where one sensitive value falls at an index ≥ 32, all other non-sensitive values beyond index 31 will now be logged as `(redacted)` too. If the intent is "fail safe by over-redacting," worth a comment explicitly calling that out in the function header; otherwise this needs per-index tracking for the tail (e.g. a small overflow list or a higher cap).

File: src/commandlog.c [hunk 2]
```diff
@@ -43,9 +43,11 @@ static commandlogEntry *commandlogCreateEntry(client *c, robj **argv, int argc,
             ce->argv[j] =
                 createObject(OBJ_STRING, sdscatprintf(sdsempty(), "... (%d more arguments)", argc - ceargc + 1));
         } else {
-            /* Trim too long strings as well... */
-            if (argv[j]->type == OBJ_STRING && sdsEncodedObject(argv[j]) &&
-                sdslen(objectGetVal(argv[j])) > COMMANDLOG_ENTRY_MAX_STRING) {
+            if (clientCommandArgShouldBeRedacted(c, j)) {
+                ce->argv[j] = shared.redacted;
```
Comment (sarthakaggarwal97) [inline src/commandlog.c:47]: The existing `COMMANDLOG slow - Certain commands are omitted...` and `MIGRATE ... AUTH` tests in `tests/unit/commandlog.tcl` only exercise single-command redaction, which works identically in both implementations. Neither the MULTI/EXEC cross-command leak nor the ≥32-index sentinel path are exercised. Given this PR is specifically reworking redaction, please add at least:

- a MULTI with a redacting command followed by a non-redacting one, asserting the second command logs unredacted,
- a command that triggers `redactClientCommandArgument` with `argc >= 32` (e.g. long `ACL SETUSER` or synthetic via module), confirming behavior.

**Review by sarthakaggarwal97 — COMMENTED** [review]
Automated review found 2 issue(s). Highest severity: `medium`.
